oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java

```diff
@@ -278,8 +279,15 @@ protected boolean indexTypeOrderedFields(Document doc, String pname, int tag, Pr
                 f = new SortedDocValuesField(name,
                         new BytesRef(property.getValue(Type.BOOLEAN).toString()));
             } else if (tag == Type.STRING.tag()) {
-                f = new SortedDocValuesField(name,
-                        new BytesRef(property.getValue(Type.STRING)));
+                String stringValue = property.getValue(Type.STRING);
+                BytesRef bytesRef;
```

Comment:
I would log the length, and simplify a bit, something like this (not tested):

    String value = property.getValue(Type.STRING);
    if (value.length() > STRING_PROPERTY_MAX_LENGTH) {
        log.warn("Truncating property {} for node {} as its length {} is > {}", 
                name, this.path, value.length(), STRING_PROPERTY_MAX_LENGTH);
        value = value.substring(0, STRING_PROPERTY_MAX_LENGTH));
    }
    f = new SortedDocValuesField(name, new BytesRef(value));

Reply:
Done